src/cljs/athens/views/blocks.cljs

```diff
@@ -609,11 +613,22 @@
                        :context-menu/x nil
                        :context-menu/y nil
                        :context-menu/show false})]
+
+    ;; If :generated-str is updated, automatically update atom-string
+    ;; Necessary because modifying generated-str itself won't trigger the on-change event of the textarea
+    ;; atom-string must be modified to trigger generated-str change
+    (add-watch state :generated-str-listener
+               (fn [_context _atom old new]
+                 (when (and (not= (:generated-str old) (:generated-str new))
+                            (not (nil? (:generated-str new))))
+                   (swap! state assoc :atom-string (:generated-str new)))))
+
```

Comment:
Add another listener so that `:search/results` are computed when `:search/query` is updated. Flowing data, FRP style.

Reply:
Not super complex, so fine to mutate `:search/query` and `:search/results` coupled